core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -640,12 +645,19 @@ private[kafka] class Processor(val id: Int,
         openOrClosingChannel(receive.source) match {
           case Some(channel) =>
             val header = RequestHeader.parse(receive.payload)
-            val context = new RequestContext(header, receive.source, channel.socketAddress,
+            val connectionId = receive.source
+            val context = new RequestContext(header, connectionId, channel.socketAddress,
               channel.principal, listenerName, securityProtocol)
+
+            // This callback is used for quota managers to throttle on quota violation by delaying the channel unmute.
+            def channelUnmuteCallback() = {
```

Comment:
I think channelUnmutingCallback would be better name

Reply:
Done